File: lib/presentation/widgets/stat_card.dart
import 'package:flutter/material.dart';

/// A compact metric card: icon, value, label and an optional trend delta.
class StatCard extends StatelessWidget {
  const StatCard({
    super.key,
    required this.icon,
    required this.value,
    required this.label,
    this.trend,
    this.color,
  });

  final IconData icon;
  final String value;
  final String label;

  /// e.g. "+12.4%". Positive -> green, leading "-" -> red.
  final String? trend;
  final Color? color;

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final accent = color ?? theme.colorScheme.primary;
    final isNegative = trend != null && trend!.trim().startsWith('-');

    return Container(
      width: 200,
      padding: const EdgeInsets.all(20),
      decoration: BoxDecoration(
        color: theme.colorScheme.surface,
        borderRadius: BorderRadius.circular(16),
        border: Border.all(color: theme.colorScheme.outlineVariant),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisSize: MainAxisSize.min,
        children: [
          Container(
            padding: const EdgeInsets.all(10),
            decoration: BoxDecoration(
              color: accent.withValues(alpha: 0.12),
              borderRadius: BorderRadius.circular(12),
            ),
            child: Icon(icon, color: accent),
          ),
          const SizedBox(height: 16),
          Text(
            value,
            style: theme.textTheme.headlineSmall?.copyWith(fontWeight: FontWeight.bold),
          ),
          const SizedBox(height: 4),
          Text(label, style: theme.textTheme.bodyMedium?.copyWith(
            color: theme.colorScheme.onSurfaceVariant,
          )),
          if (trend != null) ...[
            const SizedBox(height: 12),
            Row(
              children: [
                Icon(
                  isNegative ? Icons.trending_down : Icons.trending_up,
                  size: 16,
                  color: isNegative ? Colors.red : Colors.green,
                ),
                const SizedBox(width: 4),
                Text(
                  trend!,
                  style: TextStyle(
                    color: isNegative ? Colors.red : Colors.green,
                    fontWeight: FontWeight.w600,
                    fontSize: 13,
                  ),
                ),
              ],
            ),
          ],
        ],
      ),
    );
  }
}

File: android_native/app/src/main/java/com/example/widgetgallery/presentation/components/StatCard.kt
package com.example.widgetgallery.presentation.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.TrendingDown
import androidx.compose.material.icons.filled.TrendingUp
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

/** Compact metric card with icon, value, label and trend delta. */
@Composable
fun StatCard(
    icon: ImageVector,
    value: String,
    label: String,
    modifier: Modifier = Modifier,
    trend: String? = null,
) {
    val scheme = MaterialTheme.colorScheme
    val negative = trend?.trim()?.startsWith("-") == true
    Column(
        modifier
            .width(200.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(scheme.surface)
            .border(1.dp, scheme.outlineVariant, RoundedCornerShape(16.dp))
            .padding(20.dp),
    ) {
        Box(
            Modifier
                .clip(RoundedCornerShape(12.dp))
                .background(scheme.primary.copy(alpha = 0.12f))
                .padding(10.dp),
        ) { Icon(icon, null, tint = scheme.primary) }
        Spacer(Modifier.height(16.dp))
        Text(value, style = MaterialTheme.typography.headlineSmall,
            fontWeight = FontWeight.Bold)
        Spacer(Modifier.height(4.dp))
        Text(label, color = scheme.onSurfaceVariant,
            style = MaterialTheme.typography.bodyMedium)
        if (trend != null) {
            Spacer(Modifier.height(12.dp))
            val c = if (negative) Color(0xFFD32F2F) else Color(0xFF2E7D32)
            Row(verticalAlignment = Alignment.CenterVertically) {
                Icon(
                    if (negative) Icons.Filled.TrendingDown
                    else Icons.Filled.TrendingUp,
                    null, tint = c, modifier = Modifier.size(16.dp),
                )
                Spacer(Modifier.width(4.dp))
                Text(trend, color = c, fontWeight = FontWeight.SemiBold)
            }
        }
    }
}
